src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2257,5 +2257,100 @@ private static bool TryFindFirstMatchedLane(Vector128<byte> mask, Vector128<byte
+            {
+                Vector256<byte> reverseMask = Vector256.Create(
+                    (byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, // first 128-bit lane
+                    15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0); // first 128-bit lane
+                last = ref Unsafe.Subtract(ref Unsafe.Add(ref first, numBytes), Vector256<byte>.Count);
+                do
+                {
+                    // Load in values from beginning and end of the array.
+                    Vector256<byte> tempFirst = Unsafe.As<byte, Vector256<byte>>(ref first);
+                    Vector256<byte> tempLast = Unsafe.As<byte, Vector256<byte>>(ref last);
+
+                    // Avx2 operates on two 128-bit lanes rather than the full 256-bit vector.
+                    // Perform a shuffle to reverse each 128-bit lane, then permute to finish reversing the vector:
+                    //     +-------------------------------------------------------------------------------+
+                    //     | A1 | B1 | C1 | D1 | E1 | F1 | G1 | H1 | I1 | J1 | K1 | L1 | M1 | N1 | O1 | P1 |
+                    //     +-------------------------------------------------------------------------------+
+                    //     | A2 | B2 | C2 | D2 | E2 | F2 | G2 | H2 | I2 | J2 | K2 | L2 | M2 | N2 | O2 | P2 |
+                    //     +-------------------------------------------------------------------------------+
+                    //         Shuffle --->
+                    //     +-------------------------------------------------------------------------------+
+                    //     | P1 | O1 | N1 | M1 | L1 | K1 | J1 | I1 | H1 | G1 | F1 | E1 | D1 | C1 | B1 | A1 |
+                    //     +-------------------------------------------------------------------------------+
+                    //     | P2 | O2 | N2 | M2 | L2 | K2 | J2 | I2 | H2 | G2 | F2 | E2 | D2 | C2 | B2 | A2 |
+                    //     +-------------------------------------------------------------------------------+
+                    //         Permute --->
+                    //     +-------------------------------------------------------------------------------+
+                    //     | P2 | O2 | N2 | M2 | L2 | K2 | J2 | I2 | H2 | G2 | F2 | E2 | D2 | C2 | B2 | A2 |
+                    //     +-------------------------------------------------------------------------------+
+                    //     | P1 | O1 | N1 | M1 | L1 | K1 | J1 | I1 | H1 | G1 | F1 | E1 | D1 | C1 | B1 | A1 |
+                    //     +-------------------------------------------------------------------------------+
+                    tempFirst = Avx2.Shuffle(tempFirst, reverseMask);
+                    tempFirst = Avx2.Permute2x128(tempFirst, tempFirst, 1);
+                    tempLast = Avx2.Shuffle(tempLast, reverseMask);
+                    tempLast = Avx2.Permute2x128(tempLast, tempLast, 1);
```

Comment:
Maybe?
```suggestion
                    tempFirst = Avx2.Shuffle(tempFirst, reverseMask);
                    tempFirst = Avx2.Permute2x128(tempFirst, tempFirst, 0b00_01);
                    tempLast = Avx2.Shuffle(tempLast, reverseMask);
                    tempLast = Avx2.Permute2x128(tempLast, tempLast, 0b00_01);
```

This is a "personal preference/suggestion" here. Permute takes an immediate of the form `0bYY_XX` where `X` is `0-3` and picks the upper or lower 128-bits from `left` (`0 is `left.GetLower()`, `1` is `left.GetUpper()) or `right` (`2` is `right.GetLower()`, `1` is `right.GetUpper()`). The same for `Y`

I find explicitly calling out both halves to be helpful here and that it helps with refactorings and/or understanding the logic